HW-5. MVVM 패턴 적용

onxmoreplz opened the pull request:
recent_movie_item 레이아웃 파일과 RecentViewModel 데이터바인딩 이용해서 Click 이벤트 처리하려했으나
왠지모르게 RecentSearchViewModel에서 Observable 값들을 Set 해줘도
RecentSearchActivity에서 addOnPropertyChanged 콜백함수가 호출이 안되네요...
코드수정 가능성이 있어 주석처리 해놨습니다. 

혹시 예상되는 해결방법이 있다면 Comment 부탁드려요!

onxmoreplz committed:
HW-5. MainViewModel 클래스 생성

onxmoreplz committed:
HW-5. MainViewModel 생성에 따른 Presenter, Contract 삭제

onxmoreplz committed:
HW-5. MainViewModel 생성에 따른 DataBinding,이벤트 처리 코드 생성

onxmoreplz committed:
HW-5. main 레이아웃 파일과 MainViewModel 클래스 Data Binding

onxmoreplz committed:
HW-5. MainViewModel Optimize Import

onxmoreplz committed:
HW-5. 기존 MVP패턴의 RecentSearch Contract, Presenter 삭제

onxmoreplz committed:
HW-5. RecentSearchActivity ViewModel 바인딩을 위한 수정

- ViewModel 이벤트 콜백함수 정의
- RecentSearchRecyclerViewAdapter와 클릭리스너 통신을 위한 ClickSavedMovieListener 인터페이스 구현

onxmoreplz committed:
HW-5. ReacentSearchAdapter) ViewModel 바인딩

onxmoreplz committed:
HW-5. ReacentSearchViewModel 클래스 생성

onxmoreplz committed:
HW-5. setQueryList 바인딩어댑터 정의

함수에 List<Stiring> 전달 시, 리스트 set 시켜주는 기능

onxmoreplz committed:
HW-5. activity_main 레이아웃파일 Reformat Code

onxmoreplz committed:
HW-5. activity_recent_search 레이아웃 파일 Reformat code

onxmoreplz committed:
HW-5. recent_movie_item 레이아웃 파일과 RecentSearchViewModel과 데이터바인딩

dev-donghwan commented on class14/onxmoreplz/HW2_Project/app/src/main/java/com/example/hw2_project/main/MainViewModel.kt:
```diff
@@ -0,0 +1,41 @@
+    val movieListTest: ObservableField<List<com.example.hw2_project.data.api.NaverMovieData.NaverMovie>> = ObservableField()
+
+    val isEmptyQuery = ObservableField<Unit>()
+    val successToGetMovie = ObservableField<Unit>()
+    val failToGetMovie = ObservableField<Unit>()
```
같은 기능을 수행하지만 성공과 실패로 나뉘는데 Boolean값으로 true false를 나누는건 어떨까요?
--> 정정하겠습니다 LiveData가 아니라서 현재 방식이 더 좋을거 같아요.

onxmoreplz replied:
넵!

sjjeong commented on class14/onxmoreplz/HW2_Project/app/src/main/java/com/example/hw2_project/ViewExtensions.kt:
```diff
@@ -26,4 +27,10 @@ fun ImageView.loadImage(url: String?) {
+
+@BindingAdapter("setQueryList")
+fun RecyclerView.setRecentQueryList(recentQueryList: List<String>) {
+    var recentSearchRecyclerViewAdapter = adapter as RecentSearchRecyclerViewAdapter
```
val로 선언해주시고 as?를 사용하도록 해주세요

onxmoreplz replied:
수정했습니다.

08f6028

sjjeong commented on class14/onxmoreplz/HW2_Project/app/src/main/java/com/example/hw2_project/recentSearch/RecentSearchRecyclerViewAdapter.kt:
```diff
@@ -28,10 +25,12 @@ class RecentSearchRecyclerViewAdapter(
             LayoutInflater.from(parent.context),
             R.layout.recent_movie_item, parent, false
         )
+        binding.vm = RecentSearchViewModel(repository)
```
여기서 ViewModel를 새로 만들지 말아주세요

onxmoreplz replied:
사용되는 곳이 없었네요.

8f2a446

sjjeong commented on class14/onxmoreplz/HW2_Project/app/src/main/res/layout/activity_main.xml:
```diff
@@ -26,45 +27,45 @@
-                android:textSize="8pt"
                 android:layout_weight="1"
-                android:onClick="@{() -> mainActivity.clickSearchBtn()}"/>
+                android:onClick="@{() -> vm.getMovieFromRepository(etMovieName.getText().toString())}"
```
검색어는 파라미터로 전달하지 마시고 two-way처리 해주세요

onxmoreplz replied:
파라미터 없앤 뒤, 2-way 데이터바인딩 적용했습니다.
 
127a3ec

onxmoreplz committed:
viewModel 생성 코드 삭제

onxmoreplz committed:
ViewExtension) adapter Nullable로 변경

onxmoreplz committed:
HW-5. 2-way Databinding 방식으로 변경

1. activity_main 레이아웃) EditText값 MainViewModel에 전달
2. getMovieFromRepository함수 파라미터 없앰
3. MainViewModel에 queryObservableField 변수 추가
4. MainActivity에서 최근검색화면에서 클릭한 영화 String 값으로 set 하는 코드 추가

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋
